File: Codes in Python3/Dynamic Programming/Minimum Subset Sum difference.py
from typing import List
def minSubsetSumDifference(arr: List[str], n: int) -> int:
    total_sum=sum(arr)
    k=total_sum
    prev=[False]*(k+1)
    cur=[False]*(k+1)       
    prev[0]=cur[0]=True
    if arr[0] <= k:
        prev[arr[0]] = True
    for i in range(1,n):
        for target in range(0,k+1):
            not_take=prev[target]
            take=False
            if arr[i]<=target:
                take=prev[target-arr[i]]
            cur[target]=not_take or take
        prev=cur[:]
    mini=float('inf')
    for i in range(0,k+1):
        if prev[i]:
            s1=i
            s2=total_sum-i
            mini=min(mini,abs(s1-s2))
    return mini
    pass

File: Codes in C++/Dynamic Programming/Minimum Subset Sum difference.cpp
int minSubsetSumDifference(vector<int>& nums, int n)
{
        int total_sum=0;
        for(auto x:nums)total_sum+=x;
        int k=total_sum;
        vector<bool> prev(k+1,false);
        vector<bool> cur(k+1,false);
        prev[0]=cur[0]=true;
        prev[nums[0]]=true;
        for(int i=1;i<nums.size();i++)
        {
            for(int target=0;target<=k;target++)
            {
                bool not_take=prev[target];
                bool take=false;
                if(nums[i]<=target)
                    take=prev[target-nums[i]];
                cur[target]=not_take||take;
            }
            prev=cur;
        }
        int mini=1e7;
        for( int i=0;i<=k;i++)
        {
            int s1=i;
            int s2=total_sum-i;
            if(prev[i])
                mini=min(mini,abs(s1-s2));
        }
        return mini;
}
